### PR TITLE
Add chemicals industry (old)

### LINKED ISSUE #311: Update chemical industry demand
Update chemical industry demand
a. Get demand from JRC-IDEES
b. calculate quantity of ethylene, propylene, BTX, ammonia, and methanol (i.e. base chemicals) needed by country.
c. calculate hydrogen, CO2 and electricity demands to produce these base chemicals, based on molar masses

This issue is linked to https://github.com/calliope-project/euro-calliope/issues/102

### PR DESCRIPTION
**Pull request by tud-mchen6**
Fixes #311 .

Adding chemicals industry in the industry module.

## Checklist

Any checks which are not relevant to the PR can be pre-checked by the PR creator. All others should be checked by the reviewer. You can add extra checklist items here if required by the PR.

- [x] CHANGELOG updated
- [x] Minimal workflow tests pass
- [ ] Tests added to cover contribution (not relevant)
- [ ] Documentation updated (not relevant)
- [ ] Configuration schema updated (not relevant)

### CODE REVIEW

File: modules/industry/src/chemicals_industry.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        "Ammonia": 17,
+        "Methanol": 2,
+    }
+    molar_mass = {
+        "Ethylene": 28.05,
+        "Propylene": 42.08,
+        "BTX": 93,
+        "Ammonia": 17.01,
+        "Methanol": 32.04,
+    }
```
Comment (brynpickering) [inline modules/industry/src/chemicals_industry.py:139]: This should probably all be in the configuration

File: modules/industry/src/chemicals_industry.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
+from typing import Optional
+
+import eurocalliopelib.utils as ec_utils
+import pandas as pd
+from utils import formatting
+from utils import jrc_idees_parser as jrc
+
+H2_LHV_KTOE = 2.863  # 0.0333 TWh/kt LHV -> 2.863ktoe/kt
+MWH_PER_T_TO_KTOE_PER_KT = 0.08598  # 1 MWh/t -> 1 GWh/kt -> 0.08598 ktoe/kt
+METHANOL_LHV_KTOE = 0.476  # 19.915 MJ/kg LHV -> 19915000MJ/kt -> 0.476ktoe/kt
```
Comment (brynpickering) [inline modules/industry/src/chemicals_industry.py:10]: Move to config

Comment (brynpickering) [reply modules/industry/src/chemicals_industry.py:10]: (except the MWH_PER_T_TO_KTOE_PER_KT one)

File: modules/industry/src/chemicals_industry.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+def get_chem_demand_df(
+    year_range: list,
+    path_energy_balances: str,
+    path_cat_names: str,
+    path_carrier_names: str,
+    path_jrc_energy: str,
+    path_jrc_production: str,
+    path_output: Optional[str] = None,
+):
+    # -------------------------------------------------------------------------
```
Comment (brynpickering) [inline modules/industry/src/chemicals_industry.py:24]: Missing docstring

File: modules/industry/src/chemicals_industry.py [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+    # Prepare data files
+    # -------------------------------------------------------------------------
```
Comment (brynpickering) [inline modules/industry/src/chemicals_industry.py:26]: Split out these commented steps into separate functions

File: modules/industry/src/chemicals_industry.py [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+    )
+    filled_consumption_df = filled_consumption_df.stack()
+
+    if path_output is not None:
+        filled_consumption_df.reorder_levels(formatting.LEVEL_ORDER).to_csv(path_output)
+
+    return filled_consumption_df
+
+
+def process_chemicals_energy_consumption(electrical_consumption, prod_df, demand):
```
Comment (brynpickering) [inline modules/industry/src/chemicals_industry.py:84]: Missing type hints and docstring explain the input args

**Review by brynpickering — COMMENTED** [review]
Moved comments over from #355.

One of the main things to note here: the energy consumption is being grouped into just a few carriers. This is a strong assumption that I made as to how the chemicals industry will decarbonise (via electrification and synthetic methanol). In future, we would probably want to keep the output of this file in terms of mass of different chemicals and then have an additional step that allows for it to be reformulated to a common hydrocarbon that could be met by carbon-neutral means. As it currently stands, an `# ASSUME: ...` comment is necessary to explain the process.

**Comment by brynpickering** [comment]
@irm-codebase ready to rebase onto develop now that #340 is merged in!

**Comment by irm-codebase** [comment]
Closed since a new version of this feature was created by @tud-mchen6

**Pull request closed without merging**